lib/db.js

```diff
@@ -119,7 +119,7 @@ const legalOptionNames = [
  * @param {number} [options.bufferMaxEntries=-1] Sets a cap on how many operations the driver will buffer up before giving up on getting a working connection, default is -1 which is unlimited.
  * @param {(ReadPreference|string)} [options.readPreference] The preferred read preference (ReadPreference.PRIMARY, ReadPreference.PRIMARY_PREFERRED, ReadPreference.SECONDARY, ReadPreference.SECONDARY_PREFERRED, ReadPreference.NEAREST).
  * @param {object} [options.pkFactory] A primary key factory object for generation of custom _id keys.
- * @param {object} [options.promiseLibrary] A Promise library class the application wishes to use such as Bluebird, must be ES6 compatible
+ * @deprecated {object} [options.promiseLibrary] A Promise library class the application wishes to use such as Bluebird, must be ES6 compatible
```

Comment:
Unsure how to deprecate a parameter.

Reply:
There's a [deprecateOptions util](https://github.com/mongodb/node-mongodb-native/blob/2a3bb7cb636d1551abe0f4c3ccdde90170274ab4/lib/utils.js#L598), sample usage [here](https://github.com/mongodb/node-mongodb-native/blob/master/lib/db.js#L484)

Reply:
There's a helper, sample usage [here](https://github.com/mongodb/node-mongodb-native/blob/master/lib/db.js#L484) - looks like you just add a `DEPRECATED:` prefix in the jsdoc description text.

Reply:
Tried to implement, tests failed, my understanding is this is possible for methods, but not constructors.

Reply:
That's likely true, sorry about the misleading advice - I didn't look at it super closely. It looks like there's [another helper](https://github.com/mongodb/node-mongodb-native/blob/665b3524080776d836c69351c3db1cb7f211c3dc/lib/utils.js#L576) we could use to do a manual check in the constructor.

```
if (options.promiseLibrary) {
  emitDeprecationWarning('MongoClient option [promiseLibrary] is deprecated and will be removed in a later version.`);
}
```